lib/manageiq_performance/middlewares/active_support_timers.rb

```diff
@@ -33,28 +33,36 @@ def active_support_timers_finish(env)
                       active_support_timers_long_form_data
         end
       ensure
-        Thread.current[:miq_perf_request_timer_data] = nil
+        ActiveSupportTimers.query_data = nil
       end
 
       def active_support_timers_save event
-        Thread.current[:miq_perf_request_timer_data] = parsed_data(event)
+        ActiveSupportTimers.query_data = parsed_data(event)
       end
 
       def active_support_timers_long_form_data
-        proc { Thread.current[:miq_perf_request_timer_data] }
+        proc { ActiveSupportTimers.query_data }
       end
 
       def active_support_timers_short_form_data
         proc do
           {
-            "request"      => Thread.current[:miq_perf_request_timer_data]["path"],
-            "total_time"   => Thread.current[:miq_perf_request_timer_data]["time"]["total"],
-            "activerecord" => Thread.current[:miq_perf_request_timer_data]["time"]["activerecord"],
-            "views"        => Thread.current[:miq_perf_request_timer_data]["time"]["views"]
+            "request"      => ActiveSupportTimers.query_data["path"],
+            "total_time"   => ActiveSupportTimers.query_data["time"]["total"],
+            "activerecord" => ActiveSupportTimers.query_data["time"]["activerecord"],
+            "views"        => ActiveSupportTimers.query_data["time"]["views"]
           }
         end
       end
 
+      def self.query_data
```

Comment:
Scratch that, I guess we are hitting the module's "class_method", which again is just a wrapper, so I guess this is fine.

Still, not sure I see much of a benefit here.  We are basically adding more code for 10ish less characters when called, but requiring a more confusing interface implementation to do it.  My first thought when I saw this is "that is messing with a class variable, this isn't thread safe!", and I don't think I would be the first to think that.